Update localization cli and readme (#160)
SalesforceCommerceCloud/pwa-kit: packages/pwa/app/translations/README.md

# Localization

Most of the files in this folder are generated by `react-intl` **CLI tool**:

-   `pwa/app/translations/en.json` <- output of _extracting_ the default messages, which you can send to your translators
-   `pwa/app/translations/[locale].json` <- the files that your translators make for the other locales
-   `pwa/app/translations/compiled/[locale].json` <- output of _compiling_ the messages into AST format
    -   Compiling helps improve the performance because it allows `react-intl` to skip the parsing step

Several **npm scripts** are available to you that make it easier to use the CLI tool. See `pwa/package.json` for more details.

-   To **extract the default messages** in your app, run `npm run extract-default-translations` to have all of the default messages extracted into their own JSON file.
-   To **compile the translations** from all of the locales, run `npm run compile-translations`.
-   To run both an extract and compile, run `npm run build-translations`

## Supported Locales and Default Locale

> The default locale, suppoted locales, and currency settings are defined in `pwa/package.json` in an object called `l10n`.

Reviewer: ```suggestion
The default locale, supported locales, and currency settings are defined in `pwa/package.json` in an object called `l10n`.
```

}
```

## Default Messages for Supported Locales

> The deafult messages used by the locale selector are defined in the `_app` component:

Reviewer: ```suggestion
The default messages used by the locale selector are defined in the `_app` component:
```

    -   so you don’t have to create the JSON file yourself by hand
-   `npm run compile-translations` to compile all locales into Abstract Syntax Tree (AST) format
    -   `react-intl` can use this AST format to bypass the parsing step and thus, save a bit of time.
-   `npm run build-translations` to run both of the above
    -   run this whenever you add a new formatted message to your components to update the default locale's translations

Whenever you do a build, the extract and compile scripts are run automatically in case you forgot to extract the latest messages.

## Testing with a Pseudo Locale

To check whether you’ve wrapped all of the hardcoded strings with either `intl.formatMessage()` or `<FormattedMessage />` , there’s a quick way to test that by running `npm run start:pseudolocale`. It runs your local dev server with the locale forced to the pseudo locale.

Loading the site in your browser, you can quickly see that those messages that have been formatted have a long `SSS` appended to them.

## Summary

Putting all of the above together, the process for adding a new locale is as follows:

1. Enable the new locale in Business Manager of your B2C Commerce instance

> 2. Add the new locale to locale.config.js

Reviewer: Sorry @vcua-mobify we can no longer assume that localization.config.js is there, as I have just closed my PR #152 . See https://github.com/SalesforceCommerceCloud/pwa-kit/pull/152#issuecomment-935138511 for more details.

For your PR, can you add an additional step to this readme here? To mention about updating the `defaultLocaleMessages` too in `App`.

Afterwards, once we have the better implementation, we can go back and update this readme.

Author: @vmarta The readme has been updated to remove references to locale.config.js

We’ve provided a few npm scripts to help you:

-   `npm run extract-default-translations` to extract the default messages
    -   so you don’t have to create the JSON file yourself by hand
-   `npm run compile-translations` to compile all locales into Abstract Syntax Tree (AST) format
    -   `react-intl` can use this AST format to bypass the parsing step and thus, save a bit of time.
-   `npm run build-translations` to run both of the above

>     -   run this whenever you add a new formatted message to your components to update the default locale's translations

Reviewer: 🤔 do we want to suggest to simply have this `build-translation` as the 1 command to remember?

At the minimum, only defaultMessage is the required parameter. The message id is optional. If you don’t specify it, the id is auto-generated for you.

## Workflow Between Developers and Translation Team

During your development, you extract and compile the messages and pass the generated JSON files to the translation team, as well as using the translated files in the app.

> We’ve provided a the following npm scripts to help you:

Reviewer: ```suggestion
We’ve provided the following npm scripts to help you:
```

Loading the site in your browser, you can quickly see that those messages that have been formatted have a long `SSS` appended to them.

### Summary

Putting all of the above together, the process for adding a new locale is as follows:

> 1. Enable the new locale in your B2C Commerce instance

Reviewer: ```suggestion
1. Enable the new locale in Business Manager of your B2C Commerce instance
```

2. Add the new locale to locale.config.js
3. If the new locale is the new default:
    - Update your React components' default messages to the new default locale's language
    - In locale.config.js, set the new default locale
    - Run `npm run extract-default-translations` to extract the new default locale's translations into `pwa/app/translations/[defaultLocale].json`
    - Send the extracted translations to your translation team

> 4. Add the new locale's transtions from your translation team to `pwa/app/translations/[locale].json`

Reviewer: I think you meant _other_ locales here and not the "new locale".

```suggestion
4. Place the files you receive from your translation team into the `pwa/app/translations/` folder
```
